diff --git a/hoti2024/demo4/generate_all_reduce_128nodes.py b/hoti2024/demo4/generate_all_reduce_128nodes.py
index 4de61b4..dd1c02a 100644
--- a/hoti2024/demo4/generate_all_reduce_128nodes.py
+++ b/hoti2024/demo4/generate_all_reduce_128nodes.py
@@ -37,7 +37,7 @@ def main() -> None:
             node.attr.append(ChakraAttr(name="is_cpu_op", bool_val=False))
             node.attr.append(ChakraAttr(name="comm_type", int64_val=ALL_REDUCE))
             node.attr.append(ChakraAttr(name="comm_size", uint64_val=coll_size))
-            node.attr.append(ChakraAttr(name="involved_dim", bool_list=BoolList(values=[True])))
+            node.attr.append(ChakraAttr(name="involved_dim", bool_list=BoolList(values=[True, True])))
 
             # store Chakra ET file
             encode_message(et, node)
diff --git a/hoti2024/demo4/inputs/Ring_Switch_sys.json b/hoti2024/demo4/inputs/Ring_Switch_sys.json
index 120ca2d..9a9dadd 100644
--- a/hoti2024/demo4/inputs/Ring_Switch_sys.json
+++ b/hoti2024/demo4/inputs/Ring_Switch_sys.json
@@ -2,7 +2,7 @@
     "scheduling-policy": "LIFO",
     "endpoint-delay": 10,
     "active-chunks-per-dimension": 1,
-    "preferred-dataset-splits": 4,
+    "preferred-dataset-splits": 1,
     "all-reduce-implementation": ["ring","ring"],
     "all-gather-implementation": ["ring","ring"],
     "reduce-scatter-implementation": ["ring","ring"],
